app/controllers/financial_transactions_controller.rb

```diff
@@ -23,41 +23,33 @@ def new
     @financial_transaction.transactable_id = params[:transactable_id] if @financial_transaction.transactable_type != Payment.name
   end
 
-  # GET /financial_transactions/1/edit
-  def edit; end
-
-  # POST /financial_transactions
   def create
     @financial_transaction = FinancialTransaction.new(financial_transaction_params)
     if @financial_transaction.transactable_type == Payment.name
-      payment = Payment.new(payment_params) # hard fail
+      payment = Payment.new(payment_params)
       unless payment.save
         flash[:danger] = 'Please Properly Fill Out Contact And Payment Fields'
-        render(:new) && return
+        render :new and return
       end
+
       @financial_transaction.transactable_id = payment.id
     end
 
     if @financial_transaction.save
-      redirect_to rental_invoice_path(@financial_transaction.rental_id), success: 'Financial Transaction Successfully Created'
+      redirect_to rental_invoice_path(@financial_transaction.rental_id),
+                  success: 'Financial Transaction Successfully Created'
```

Comment:
Shouldn't this be in the flash?

Reply:
Yes, that will also be addressed in #382